main.js

```diff
@@ -0,0 +1,12 @@
+var parser = require('./parser.js').parser,
+    makeXPathModels = require('./models.js').makeXPathModels;
+
+parser.makeXPathModels = makeXPathModels;
+
+parser.setXPathModels = function(models) {
+    parser.yy.xpathmodels = models;
```

Comment:
You could set up new parsers before, but since it used global state for xpathmodels it was pretty much useless